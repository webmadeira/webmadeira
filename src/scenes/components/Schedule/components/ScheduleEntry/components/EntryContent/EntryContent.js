import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import {
  Subtitle1,
  Caption,
  Body1,
  Body2,
} from '../../../../../../../components/typography'
import deviceWidths from '../../../../../../../theme/device-widths'

const ENTRY_TYPES = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
}

const Root = styled.div`
  margin-bottom: 35px;
`

const EntryTitle = styled(Subtitle1)`
  margin-bottom: 10px;
  position: relative;
`

const EntryBody = styled(Subtitle1)`
  margin-bottom: 12px;
`

const LengthLabel = styled(Caption)`
  color: ${props => props.theme.color('secondary', 'light')};
`

const Bullet = styled.div`
  ${({ theme, type }) => {
    let size
    let positionLeft
    let positionTop
    let color

    if (type === ENTRY_TYPES.PRIMARY) {
      size = 24
      positionLeft = -56
      positionTop = -2
      color = theme.color('brand', 'light')
    } else {
      size = 12
      positionLeft = -50
      positionTop = 4
      color = theme.color('neutral', 'bright')
    }

    return `
      position: absolute;
      bottom: 0;
      left: ${positionLeft}px;
      top: ${positionTop}px;
      display: block;
      width: ${size}px;
      height: ${size}px;
      content: "";
      background-color: ${color};
      border: 2px solid ${theme.color('neutral', 'bright')};
      border-radius: ${size}px;

      @media (max-width: ${deviceWidths.phone / 16}em) {
        width: ${size - 6}px;
        height: ${size - 6}px;
        left: ${positionLeft + 28}px;
      }
    `
  }};
`

const EntryContent = ({
  title,
  type,
  speakers,
  description,
  length,
  qaLength,
}) => {
  const entryType = (type === 'talk' || type === 'primary') ? ENTRY_TYPES.PRIMARY : ENTRY_TYPES.SECONDARY

  const speakerInfo = speakers && (
    <EntryBody>
      <Body1>{speakers.join(', ')}</Body1>
      <Body2>{description}</Body2>
    </EntryBody>
  )

  let timeInfo

  if (length && qaLength) {
    timeInfo = (<LengthLabel>{length}min + {qaLength}min</LengthLabel>)
  } else if (length) {
    timeInfo = (<LengthLabel>{length}min</LengthLabel>)
  }

  return (
    <Root>
      <EntryTitle>
        <Bullet type={entryType} />
        {title}
      </EntryTitle>
      {speakerInfo}
      {timeInfo}
    </Root>
  )
}

EntryContent.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string,
  speakers: PropTypes.arrayOf(PropTypes.string),
  description: PropTypes.string,
  length: PropTypes.number,
  qaLength: PropTypes.number,
}

EntryContent.defaultProps = {
  speakers: undefined,
  description: undefined,
  length: undefined,
  qaLength: undefined,
  type: undefined,
}

export default EntryContent
