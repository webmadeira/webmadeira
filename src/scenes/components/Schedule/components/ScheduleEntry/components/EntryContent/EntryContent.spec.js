import React from 'react'
import { shallow } from 'enzyme'
import EntryContent from './EntryContent'

describe('EntryContent component', () => {
  let title
  let speaker
  let description
  let length
  let qaLength

  beforeEach(() => {
    title = 'Title'
    speaker = 'Speaker name'
    description = 'Some description'
    length = 20
    qaLength = 5
  })

  describe('with time and QA', () => {
    it('must render properly', () => {
      const component = shallow(
        <EntryContent
          title={title}
          speaker={speaker}
          description={description}
          length={length}
          qaLength={qaLength}
        />,
      )

      expect(component).toMatchSnapshot()
    })
  })

  describe('with no QA', () => {
    it('must render properly', () => {
      const component = shallow(
        <EntryContent
          title={title}
          speaker={speaker}
          description={description}
          length={length}
        />,
      )

      expect(component).toMatchSnapshot()
    })
  })

  describe('with no time', () => {
    it('must render properly', () => {
      const component = shallow(
        <EntryContent
          title={title}
          speaker={speaker}
          description={description}
        />,
      )

      expect(component).toMatchSnapshot()
    })
  })
})
