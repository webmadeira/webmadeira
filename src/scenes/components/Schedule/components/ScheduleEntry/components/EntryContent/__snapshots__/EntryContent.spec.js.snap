// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryContent component with no QA must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EntryContent
    description="Some description"
    length={20}
    qaLength={undefined}
    speaker="Speaker name"
    speakers={undefined}
    title="Title"
    type={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(styled.h4)>
          <styled.div
            type="secondary"
          />
          Title
        </Styled(styled.h4)>,
        undefined,
        <Styled(styled.span)>
          20
          min
        </Styled(styled.span)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              type="secondary"
            />,
            "Title",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "type": "secondary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Title",
        ],
        "type": [Function],
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            20,
            "min",
          ],
        },
        "ref": null,
        "rendered": Array [
          20,
          "min",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(styled.h4)>
            <styled.div
              type="secondary"
            />
            Title
          </Styled(styled.h4)>,
          undefined,
          <Styled(styled.span)>
            20
            min
          </Styled(styled.span)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div
                type="secondary"
              />,
              "Title",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "type": "secondary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Title",
          ],
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              20,
              "min",
            ],
          },
          "ref": null,
          "rendered": Array [
            20,
            "min",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`EntryContent component with no time must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EntryContent
    description="Some description"
    length={undefined}
    qaLength={undefined}
    speaker="Speaker name"
    speakers={undefined}
    title="Title"
    type={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(styled.h4)>
          <styled.div
            type="secondary"
          />
          Title
        </Styled(styled.h4)>,
        undefined,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              type="secondary"
            />,
            "Title",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "type": "secondary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Title",
        ],
        "type": [Function],
      },
      undefined,
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(styled.h4)>
            <styled.div
              type="secondary"
            />
            Title
          </Styled(styled.h4)>,
          undefined,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div
                type="secondary"
              />,
              "Title",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "type": "secondary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Title",
          ],
          "type": [Function],
        },
        undefined,
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`EntryContent component with time and QA must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EntryContent
    description="Some description"
    length={20}
    qaLength={5}
    speaker="Speaker name"
    speakers={undefined}
    title="Title"
    type={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(styled.h4)>
          <styled.div
            type="secondary"
          />
          Title
        </Styled(styled.h4)>,
        undefined,
        <Styled(styled.span)>
          20
          min + 
          5
          min
        </Styled(styled.span)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              type="secondary"
            />,
            "Title",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "type": "secondary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Title",
        ],
        "type": [Function],
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            20,
            "min + ",
            5,
            "min",
          ],
        },
        "ref": null,
        "rendered": Array [
          20,
          "min + ",
          5,
          "min",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(styled.h4)>
            <styled.div
              type="secondary"
            />
            Title
          </Styled(styled.h4)>,
          undefined,
          <Styled(styled.span)>
            20
            min + 
            5
            min
          </Styled(styled.span)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div
                type="secondary"
              />,
              "Title",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "type": "secondary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Title",
          ],
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              20,
              "min + ",
              5,
              "min",
            ],
          },
          "ref": null,
          "rendered": Array [
            20,
            "min + ",
            5,
            "min",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
